R/fit.R

```diff
@@ -12,8 +12,9 @@ CmdStanFit <- R6::R6Class(
     },
 
     save_object = function(file, ...) {
-      # CmdStanMCMC has its own implementation
       self$draws()
+      try(self$sampler_diagnostics(), silent = TRUE)
+      try(self$init(), silent = TRUE)
```

Comment:
We can use `try(..., silent=TRUE)` here to make sure we try to bring in everything before saving but don't error if it's not available. Doing this also for sampler_diagnostics means we don't need a separate method for CmdStanMCMC anymore.

Reply:
Nice!